### ISSUE TITLE
Installation Problem: Cannot find module Yarn-Berry

### ISSUE THREAD

**Issue by yepMad** [yarnpkg/yarn#7899]
```
C:\Windows\system32>yarn -v
internal/modules/cjs/loader.js:796
    throw err;
    ^

Error: Cannot find module 'C:\Users\eudes\.yarn\releases\yarn-berry.js'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:793:17)
    at Function.Module._load (internal/modules/cjs/loader.js:686:27)
    at Function.Module.runMain (internal/modules/cjs/loader.js:1043:10)
    at internal/main/run_main_module.js:17:11 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
```

**Comment by redblue9771** [comment]
Try uninstalling the current yarn, then `npm install -g yarn@berry` .

**Comment by regevbr** [comment]
@redblue9771 this doesn't solve the issue... can you please advise?

**Comment by redblue9771** [comment]
> @redblue9771 this doesn't solve the issue... can you please advise?

Now the official recommendation is to install Yarn v1, and then use the command to change v1 to v2. Try it

**Comment by gferreri** [comment]
I'm having this problem in a CI environment. It seems that the presence of .yarnrc that says to use berry without the downloaded release actually present breaks it. Seems like a catch-22. Is there a workaround I'm not aware of? I'd hate to commit the berry.js release to source but it seems like the only viable path...

**Comment by regevbr** [comment]
@gferreri just remove all yarn 2 related filed, then install it again using yarn1
Make sure you commit all the relevant files (e.g. .yarn/cache|plugins|releasse and .yarnrc.yml and .yarnrc) and it should not happen again.
Also make sure you have the latest yarn 1 version (1.22.4)

**Comment by Hallo89** [comment]
I have the exact same problem on my CI (Travis CI), which uses Yarn v1.15.2. Here's what happens:

Yarn 1.18 added reading `yarnPath` from v2 `.yarnrc.yml` files (see https://github.com/yarnpkg/yarn/blob/master/CHANGELOG.md#1180). This means that a global Yarn 1.18+ installation will automatically recognize a (local) v2 project as v2 and use it accordingly (via `.yarn/releases/<whatever your yarnPath points to>`)

So an installation of v1.17 and below will not recognize the v2 project correctly. Running `yarn policies set-version berry` will create a v1 `.yarnrc` lockfile which sets `yarn-path` to `.yarn/releases/yarn-berry.js`. This setup will work if the corresponding v2 project uses this exact `yarn-berry.js` file in `.yarnrc.yml`.
However, if the project has run `yarn set version latest` at least once, it replaces the berry-file with a file named after the actual version, like `yarn-2.0.0-rc.33.js`. This will break the setup, because .yarnrc (which is the only file the v1.17- Yarn can read) demands a `yarn-berry.js` which does not exist.

TL;DR: The solution:
- Upgrading the global Yarn installation to the latest version (or at least 1.18+) (which will not create & demand a `.yarnrc` file)
- Dynamically modifying `.yarnrc` to have `yarn-path` pointing to the same value as `yarnPath` in the v2 `.yarnrc.yml`

If I got anything wrong, correct me please, but I hope this helps to understand the issue a bit. (Note: at the time of this writing I have not yet solved the issue on Travis)

**Comment by nfantone** [comment]
A guide on what files are actually needed to be able to run `yarn` 2 on a per-project basis would be nice. There's absolutely no mention of that in [the current docs](https://yarnpkg.com/getting-started/install). Also, pushing `.yarn/cache` to the repository, a directory with literally thousands of files, seems like an overkill to me.

**Comment by tr00st** [comment]
Having run into the same issue... just to confirm - am I right in understanding that it's expected to commit every used version of yarn v2 into the repo's history? (and with the best-practise of storing .yan/cache, that it'd be expected to store all historically used versions of every used package in the repo's history?)

Minor note - whilst trying to work through this issue myself, I got caught up for quite a while wondering why `yarn set version 2.1.1` wouldn't work, only to discover that it requires installing a berry version first - might it be worth updating the docs for cli/set/version to note that specific versions can't be installed from v1.x - or at least linking to the installation docs that contain some of this information?

**Comment by ScottPierce** [comment]
I added the following to my `.gitignore`. It seems to be working for me:
```
/.yarn/*
!/.yarn/plugins
!/.yarn/releases
```

I also found that you need to have your `yarnPath` set in your `.yarnrc.yml`. I'm not sure how it ended up not being set for me,  but I did.
```
yarnPath: .yarn/releases/yarn-2.2.2.cjs

plugins:
  - path: .yarn/plugins/@yarnpkg/plugin-typescript.cjs
    spec: "@yarnpkg/plugin-typescript"
```

**Comment by kennethlynne** [comment]
When I upgraded and ran this on the CI it would be nice if it hinted about this .gitignore change! :) 

After making sure to check in releases it works again, but it was pretty counter-intuitive to check in what seems like a cache file

**Comment by merceyz** [comment]
https://yarnpkg.com/getting-started/install#per-project-install specifies that it should be commited so closing as this is documented
